src/alr/alr-commands-init.adb

```diff
@@ -121,56 +125,113 @@ package body Alr.Commands.Init is
       -- Generate_Project_File --
       ---------------------------
 
-      procedure Generate_Project_File is
+      procedure Generate_Project_File
+      is
+         procedure NL
+           renames New_Line;
+
+         package GPR_File
+           is new GPR_Generator (Put      => Put,
+                                 New_Line => NL);
+         use GPR_File.Shortcuts;
+
          Filename : constant String :=
             +Full_Name (Directory / (+Lower_Name & ".gpr"));
+
+         Build_Profile : constant String
+           := """obj/""" & " & " &
+                  Mixed_Name & "_Config.Build_Profile";
+
+         Library_Name : constant String
+           := "Project'Library_Name & "".so."" & "
+              & Mixed_Name & "_Config.Crate_Version";
+
+         Reloc  : constant String := """relocatable""";
+         Static : constant String := """static""";
+         PIC    : constant String := """static-pic""";
+
+         Library_Type : constant String
+           := "(" & Reloc & ", " & Static & ", " & PIC & ");";
+
+         External_1 : constant String
+           := "external (""LIBRARY_TYPE"", ""static"")";
+
+         External_2 : constant String
+           := "external (""" & Upper_Name & "_LIBRARY_TYPE"", "
+              & External_1 & ");";
       begin
-         --  Use more than 80 columns for more readable strings
-         pragma Style_Checks ("M200");
+         pragma Style_Checks ("-3");
+         --  Suspend eindention checks to let program structure
+         --  be like GPR file structue.
 
          --  Main project file
          if not Create (Filename) then
             Trace.Warning ("Cannot create '" & Filename & "'");
             return;
          end if;
-         Put_Line ("with ""config/" & Lower_Name & "_config.gpr"";");
-         Put_Line ("project " & Mixed_Name & " is");
-         Put_New_Line;
+
+         GPR_File.NL_Before_IS      := False;
+         GPR_File.NL_After_FOR_List := False;
+         --  GPR File Style
+
+         NL;
+         WC ("config/" & Lower_Name & "_config.gpr");
+         NL;
+         RB (Mixed_Name, GPR_File.Library);
+         NL;
```

Comment:
I'm not sure I like this kind of super-telegraphic style :/

Reply:
Me neither. I don't find this version to be more readable than the existing one.

In the grand scheme of things, it could be a good idea to make a library to facilitate the generation of GPR files.
But for the limited use we have in Alire, I don't think this change make sense.

Reply:
> I'm not sure I like this kind of super-telegraphic style :/

I believe it is called *Abstraction*, a feature supported by Ada 83 ;-)

Reply:
Seems more like obfuscation to me.

Reply:
> Seems more like obfuscation to me.

Then *Rapid Applcation Development* is obscure.. ;-)

Reply:
Neither abstraction nor RAD is about abbreviating subprogram names. 😉

The problem is, after a while `CO`,`EL`, `EX`, `FR`, `FU`, `LO`, `NL`, `PB`, `PU`, `QE`, `RB`, `RE`, `WC` will require jumping to their respective definitions, probably even for the author of the code, in order to understand what is happening.

The code seems to be optimized for typing time instead of comprehension time and I think that in general it's better to do the opposite.

Reply:
> Neither abstraction nor RAD is about abbreviating subprogram names. 😉

Well it might just be a *Ruby* thing then :-)

> The problem is, after a while `CO`,`EL`, `EX`, `FR`, `FU`, `LO`, `NL`, `PB`, `PU`, `QE`, `RB`, `RE`, `WC` will require jumping to their respective definitions, probably even for the author of the code, in order to understand what is happening.

Well, I guess a modern IDE should be able to help you out on this one :-)

> The code seems to be optimized for typing time instead of comprehension time and I think that in general it's better to do the opposite.

Give me a break! If I optimized for typing time I wouldn't have created the PR in the first place :-) ;-)

I tried to do some data-centric approach.
I'll figure out something ..

I'be back in a day or two...

I the meantime hang on with a little entertainment [Calby - Burnout](https://youtu.be/jDrUwP09w74)

Reply:
> Well it might just be a _Ruby_ thing then :-)

I'd still be surprised, but then again, I don't know Ruby ecosystem. 😉

> Well, I guess a modern IDE should be able to help you out on this one :-)

I expected this. 😜 But it misses the point, because I'd still be forced to jump to definitions or use some kind of tooltips instead of just reading `Comment`, `Expression` etc.

> Give me a break! If I optimized for typing time I wouldn't have created the PR in the first place :-) ;-)

Ha! 😄

> I tried to do some data-centric approach.

Well, SQL is data-centric and yet we have `SELECT`, `FROM`, `WHERE` instead of `SE`, `FR`, `WH`. 😉

> I'll figure out something ..
>
> I'be back in a day or two...
>
> I the meantime hang on with a little entertainment [tralalalulalaj](https://youtu.be/jDrUwP09w74)

Thanks and have a nice time! 🙂

Reply:
> Well it might just be a _Ruby_ thing then :-)

I'd still be surprised, but then again, I don't know Ruby ecosystem. 😉

> Well, I guess a modern IDE should be able to help you out on this one :-)

I expected this. 😜 But it misses the point, because I'd still have to jump to definitions or use some sort of tooltips instead of just reading `Comment`, `Expression` etc.

> Give me a break! If I optimized for typing time I wouldn't have created the PR in the first place :-) ;-)

Ha! 😄

> I tried to do some data-centric approach.

I think it's good to look at how data-centric languages do it. For instance, SQL has `SELECT`, `FROM`, `WHERE` and not `SE`, `FR`, `WH`.

> I'll figure out something ..
>
> I'be back in a day or two...
>
> I the meantime hang on with a little entertainment [tralalalulalaj](https://youtu.be/jDrUwP09w74)

Thanks and I wish you a nice time! 🙂